app/assets/javascripts/attractions.js

```diff
@@ -0,0 +1,39 @@
+$(document).ready(function(){
+
+  $('#add-attraction').click(function(event){
+    event.preventDefault();
+
+    var url = $(this).attr('href');
+
+    $.ajax({
+      url: url,
+      type: "get",
+      beforeSend: function(xhr) { xhr.setRequestHeader('X-CSRF-Token', $('meta[name=csrf-token]').attr('content'))},
+      dataType: 'json'
+    }).done(function(data){
+        loadMarkers(data)
+    });
+
+  });
+
+});
+
+
+function loadMarkers(markerObjects){
+
+    $.each(markerObjects, function(i,item){
+      loadMarker(item);
+    });
+  }
+
+function loadMarker(markerObject){
+
+  var latitude = markerObject.longlat.coordinates[1];
+  var longitude = markerObject.longlat.coordinates[0];
+  var coords = new google.maps.LatLng(latitude, longitude)
+
+  var marker = new google.maps.Marker({
+    position: coords,
+    map: map
+  });
```

Comment:
I'd love to see this written OO style.

Perhaps you'd have a `Marker` constructor function object and a `MarkerCollection` object.  Clicking 'Add Attraction` would call `mc = new MarkerCollection(url)` which in turn would make an ajax call and then create a new Marker object.